src/stssession/STSSessionController.ts

```diff
@@ -103,39 +123,26 @@ export class STSSessionController {
       this.allowDownload = true;
     }
 
-    // Set http headers
-    var httpHeaders;
-    if (this.stsHeaderName === undefined || this.stsHeaderValue === undefined) {
-      httpHeaders = {
-        headers: {
-          'Content-Type': 'application/json'
-        }
-      };
-    } else {
-      var name = this.stsHeaderName;
-      httpHeaders = {
-        headers: {
-          'Content-Type': 'application/json',
-          name: this.stsHeaderValue
-        }
-      };
+    // set extra http headers
+    if (!this.stsHeaderName === undefined && !this.stsHeaderValue === undefined) {
+      this.httpHeaders.headers[this.stsHeaderName] = this.stsHeaderValue;
     }
 
-    var result: ng.IPromise<any> = this.httpService.post(
-      this.stsApiUrl,
-      {},
-      httpHeaders)
-    .then((response: any): ng.IPromise<any> => this.refreshSTSCredentials(response));
+    this.credentialsFromAPI();
   }
 
   /**
-   * Set the aws credentials from a credential json.
+   * Calls the sts api, and sets the aws credentials from the result.
    */
-  public refreshSTSCredentials(response: any): any {
-    this.awsAccessKeyId = response.data.AccessKeyId;
-    this.awsSecretAccessKey = response.data.SecretAccessKey;
-    this.awsSessionToken = response.data.SessionToken;
-
-    return response.data;
+  public credentialsFromAPI(): void {
+    this.httpService.post(
+      this.stsApiUrl,
+      {},
+      this.httpHeaders)
+    .then((response: any): void => {
```

Comment:
According to @CtrlC-Root's previous reviews you shouldn't use 'any' here since you should know what type to expect.

Reply:
We may not have a choice with this use of any.  `response` is an http response, and our current version of angular (1.X) does not have enhanced http handling.

Reply:
I guess it's not easy trying to figure out what the type of the response is for AngularJS since there isn't exactly a ts file that it defines. I'm OK with this then @CtrlC-Root thoughts?

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/angular/index.d.ts#L1503

Could the definition be in here somewhere?

Reply:
Found the definition from there; thanks!